lib/util/db.js

```diff
@@ -239,7 +241,65 @@ const queryFuncs = (db, obj) => {
     for (let i = 0; i < xs.length; i += 1) result[i] = f(xs[i]);
     return result;
   };
-  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, resolve).catch(reject1));
+
+  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, stream => {
+    try {
+      const twoMinutes = 2 * 60 * 1000;
+      const streamTimeout = setTimeout(() => {
+        stream.destroy(new Error('Stream timed out.'));
```

Comment:
@alxndrsn [has](https://github.com/getodk/central-backend/pull/566/files#diff-36d887fd172376101537bad122756c3090c233494f460ef975c1f4ad0fb9868aR244-R249) [mentioned](https://github.com/gajus/slonik/pull/360#issuecomment-1206303334) that passing an error to `stream.destroy()` can sometimes be an issue. However, he says that the error is infrequent and (IIRC) that if it is thrown, that could provide information to us.